components/google-cloud/tests/aiplatform/test_componets_compile.py

```diff
@@ -281,25 +272,58 @@ def pipeline():
                 project=self._project,
                 display_name=self._display_name,
                 serving_container_image_uri=self._serving_container_image_uri,
-                artifact_uri=self._artifact_uri
-            )
+                artifact_uri=self._artifact_uri)
 
             endpoint_create_op = EndpointCreateOp(
-                project=self._project, display_name=self._display_name
-            )
+                project=self._project, display_name=self._display_name)
 
             model_deploy_op = ModelDeployOp(
-                project=self._project, model=model_upload_op.outputs["model"]
-            )
+                project=self._project, model=model_upload_op.outputs["model"])
 
             batch_predict_op = ModelBatchPredictOp(
                 project=self._project,
                 model=model_upload_op.outputs["model"],
                 job_display_name=self._display_name,
                 gcs_source=self._gcs_source,
-                gcs_destination_prefix=self._gcs_destination_prefix
-            )
+                gcs_destination_prefix=self._gcs_destination_prefix)
+
+        compiler.Compiler().compile(
+            pipeline_func=pipeline, package_path=self._package_path)
+
+    def test_model_upload_op_compile(self):
+
+        @kfp.dsl.pipeline(name="training-test")
```

Comment:
I was wondering if this test would be better suited as an integration test. My thought on this is that this both testing the gcpc component as well as the kfp compiler for it. I completely agree that this test is needed though I'm wondering if we should mix unit tests with integration tests.